nix/supervisord-cluster/clusterlib.py

```diff
@@ -348,7 +348,11 @@ def send_tx_genesis(
                 f"txins: {txins} txouts: {txouts} signing keys: {signing_keys}\n{err}"
             )
 
-    def submit_update_proposal(self, cli_args, epoch=1):
+    def get_current_epoch_no(self):
+        tip = self.get_tip()
+        return int(tip["slotNo"] / self.epoch_length)
```

Comment:
Note that this is not necessarily the _current_ epoch number. The tip represents the last block that was successfully applied to the ledger, so this method actually returns the epoch number of the last adopted block. This method may return an incorrect result as there could very well be empty slots following the last adopted block.

Reply:
I think that, for the time being, I would leave this as it was prior to `e65a66df7d6ad575adc5499ad0c030397a4bf1e0`. We currently don't expose a CLI utility for retrieving the current slot number.

Reply:
Thanks for explanation, I reverted the change.